MeetupPlatformApi/Authentication/AuthenticationManager.cs

```diff
@@ -21,27 +21,43 @@ public CurrentUserInfo GetCurrentUserInfo(ClaimsPrincipal user)
         var idClaim = user.Claims.Single(claim => claim.Type == ClaimTypes.NameIdentifier);
         var id = Guid.Parse(idClaim.Value);
 
-        return new() {UserId = id};
+        return new() { UserId = id };
     }
 
-    public string IssueAccessToken(UserEntity user) =>
-        IssueToken(
+    public TokenPair IssueTokenPair(UserEntity user)
+    {
+        var accessToken = IssueAccessToken(
             payload: new Dictionary<string, object>
-            {
-                {ClaimTypes.NameIdentifier, user.Id}
-            },
-            lifetime: configuration.AccessTokenLifetime);
+                {
+                    {ClaimTypes.NameIdentifier, user.Id}
+                },
+                accessTokenLifetime: configuration.AccessTokenLifetime);
+
+        var refreshToken = IssueRefreshToken(configuration.RefreshTokenLifetime, user.Id);
 
-    private string IssueToken(IDictionary<string, object> payload, TimeSpan lifetime)
+        return new() { AccessToken = accessToken, RefreshToken = refreshToken };
+    }
+
+    private string IssueAccessToken(IDictionary<string, object> payload, TimeSpan accessTokenLifetime)
     {
         var tokenDescriptor = new SecurityTokenDescriptor
         {
             Claims = payload,
-            Expires = DateTime.UtcNow.Add(lifetime),
+            Expires = DateTime.UtcNow.Add(accessTokenLifetime),
             SigningCredentials = configuration.SigningCredentials
         };
 
         var token = tokenHandler.CreateToken(tokenDescriptor);
         return tokenHandler.WriteToken(token);
     }
+
+    private RefreshTokenEntity IssueRefreshToken(TimeSpan refreshTokenLifetime, Guid userId)
```

Comment:
This method should not create a new `RefreshTokenEntity` but issue a new encoded refresh token (as name states). It should be completely decoupled from our entities.